### DISCUSSION TITLE
Does AssetRipper has the ability to dump all `*.cs` files given `libil2cpp.so` and `global-metadata.dat`?

### DISCUSSION THREAD

**Discussion by RaenonX** [AssetRipper/AssetRipper#1475] [Q&A] [answered]
Something similar to `[Il2CppDumper.exe](https://github.com/Perfare/Il2CppDumper)`. If so, how to?
Since `Il2CppDumper` only supports up until 2022.2 and the game I want to dump is 2022.3...I have to find something else that could help dumping the `*.cs` files.

**Comment by ds5678** [comment]
Yes, it already does?

**Reply by RaenonX** [reply]
How do I let it load? Is it GUI only or CLI-doable?

**Reply by ds5678** [reply]
What CLI?

**Reply by RaenonX** [reply]
Just curious if there's an existed usage available, like `AssetRipper.exe libil2cpp.so global-metadata.dat`. Since the `exe` name includes `GUI`, I guess not.

For GUI loading the 2 files mentioned in the title though, how would I load them using GUI?

**Reply by ds5678** [reply]
It's supposed to be automatic. You give it the game, and it finds the files on its own.

**Reply by RaenonX** [reply]
By "give it the game," is it by supplying the APK file and load it as a single file?
![image](https://github.com/user-attachments/assets/703bec87-7e00-4117-b5cf-f20fc96851a7)

If so, could you point me again in the source code on where does the decompilation start? I have a repo storing `libil2cpp.so` & `global-metadata.dat` only, and would rather just load these 2 files and get the C# dumps, if possible, so I can only store 30 MB (2 files) instead of 300 MB (the whole APK with 2 versions of libil2cpp.so).

**Reply by ds5678** [reply, marked as answer]
https://github.com/SamboyCoding/Cpp2IL
